src/pykeen/training/training_loop.py

```diff
@@ -189,22 +229,76 @@ def train(
         # In some cases, e.g. using Optuna for HPO, the cuda cache from a previous run is not cleared
         torch.cuda.empty_cache()
 
-        result = self._train(
-            num_epochs=num_epochs,
-            batch_size=batch_size,
-            slice_size=slice_size,
-            label_smoothing=label_smoothing,
-            sampler=sampler,
-            continue_training=continue_training,
-            only_size_probing=only_size_probing,
-            use_tqdm=use_tqdm,
-            use_tqdm_batch=use_tqdm_batch,
-            tqdm_kwargs=tqdm_kwargs,
-            stopper=stopper,
-            result_tracker=result_tracker,
-            sub_batch_size=sub_batch_size,
-            num_workers=num_workers,
-        )
+        # A checkpoint root is always created to ensure a fallback checkpoint can be saved
+        if checkpoint_directory is None:
+            checkpoint_directory = PYKEEN_DEFAULT_CHECKPOINT_DIR
+        checkpoint_directory = pathlib.Path(checkpoint_directory)
+        checkpoint_directory.mkdir(parents=True, exist_ok=True)
+        logger.debug('using checkpoint_root at %s', checkpoint_directory)
+
+        # If a checkpoint file is given, it must be loaded if it exists already
+        save_checkpoints = False
+        checkpoint_path = None
+        if checkpoint_name:
```

Comment:
can there be a reasonable default for checkpoint_name, then like in the helper method, just pass a flag `save_checkpoints=True` so we don't have to use this logic?

Reply:
Imo it is cleaner to follow the logic that explicitly giving a `checkpoint_name` indicates that someone wants to save/reload that specific checkpoint.

Reply:
Then why not just have the user pass the fully formed path instead of both the directory and the name?